src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+import java.util.regex.Pattern;
+
+import static duke.common.Messages.MESSAGE_INVALID_COMMAND;
+
+/**
+ * Parses user input into commands.
+ */
+public class Parser {
+    private static final Pattern COMMAND_FORMAT = Pattern.compile("^(?<command>\\S+)(?<arguments>.*)$");
+
+    public static Todo parseTodo(String args) {
+        return new Todo(args);
+    }
+
+    /**
+     * Parses a deadline input string into a Deadline object.
+     *
+     * @param args the input string for the deadline
+     * @return the parsed Deadline object
+     * @throws DukeParseException if unable to parse correctly
+     */
+    public static Deadline parseDeadline(String args) throws DukeParseException {
+        Pattern pattern = Pattern.compile("^(?<desc>.+) /by (?<by>.+)$");
+        Matcher matcher = pattern.matcher(args);
+        if (!matcher.matches()) {
+            throw new DukeParseException("Deadline cannot be parsed");
+        }
+        String desc = matcher.group("desc").trim();
+        String by = matcher.group("by").trim();
+        try {
+            LocalDate byDate = LocalDate.parse(by);
+            return new Deadline(desc, byDate);
+        } catch (DateTimeParseException e) {
+            throw new DukeParseException("Deadline cannot be parsed");
+        }
+    }
+
+    /**
+     * Parses an event input string into an Event object.
+     *
+     * @param args the input string for the event
+     * @return the parsed Event object
+     * @throws DukeParseException if unable to parse correctly
+     */
+    public static Event parseEvent(String args) throws DukeParseException {
+        Pattern pattern = Pattern.compile("^(?<desc>.+) /from (?<from>.+) /to (?<to>.+)$");
+        Matcher matcher = pattern.matcher(args);
+        if (!matcher.matches()) {
+            throw new DukeParseException("Event cannot be parsed");
+        }
+        String desc = matcher.group("desc").trim();
+        String from = matcher.group("from").trim();
+        String to = matcher.group("to").trim();
+        try {
+            LocalDate fromDate = LocalDate.parse(from);
+            LocalDate toDate = LocalDate.parse(to);
+            return new Event(desc, fromDate, toDate);
+        } catch (DateTimeParseException e) {
+            throw new DukeParseException("Event cannot be parsed");
+        }
+    }
+
+    /**
+     * Takes a given user input and parses it into the corresponding command.
+     * If the user supplies invalid input, an InvalidCommand will be returned.
+     *
+     * @param input the user input
+     * @return the corresponding command from the input
+     */
+    public Command parseCommand(String input) {
+        Matcher matcher = COMMAND_FORMAT.matcher(input);
+        if (!matcher.matches()) {
+            return new InvalidCommand(MESSAGE_INVALID_COMMAND);
+        }
+        String command = matcher.group("command").trim();
+        String args = matcher.group("arguments").trim();
+
+        switch (command) {
```

Comment:
Based on the SE-EDU Coding Standard, a Basic Coding Standard rule the `case`s in a switch-case statement should be on the same indentation level as the `switch` statement